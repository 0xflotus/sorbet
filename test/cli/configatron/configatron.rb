# typed: true
class TestConfigatron
  def configatron_test
      T.assert_type!(configatron.test_bool, T.any(TrueClass, FalseClass))
      T.assert_type!(configatron.test_int, Integer)
      T.assert_type!(configatron.test_float, Float)
      T.assert_type!(configatron.test_string, String)
      T.assert_type!(configatron.test_nested.test_int, Integer)

      T.assert_type!(configatron, Configatron::RootStore)
      T.assert_type!(configatron.test_nested, Configatron::Store)

      T.assert_type!(configatron.test_nested.key?(:junk), T.any(TrueClass, FalseClass))
      T.assert_type!(configatron.key?(:junk), T.any(TrueClass, FalseClass))
  end

  def lub_test(param)
    config = if param
      configatron.test_hash_a
    else
      configatron.test_hash_b
    end

    if config && 4
      "foo"
    end
    "bar"
  end
end
