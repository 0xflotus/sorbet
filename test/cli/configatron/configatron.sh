#!/bin/bash

main/sorbet --configatron-file=test/cli/configatron/configatron.yaml test/cli/configatron/configatron.rb 2>&1
