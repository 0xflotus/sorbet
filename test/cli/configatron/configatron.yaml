default: &default
  test_bool: true
  test_int: 42
  test_string: Mamba
  test_float: 42.42
  test_nested:
      test_int: 43
