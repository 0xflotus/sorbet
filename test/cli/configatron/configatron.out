test/cli/configatron/configatron.rb:21: This code is unreachable http://go/e/7006
    21 |    if config && 4
               ^^^^^^
